tizen-api/src/tizen-api-pipeline.c

```diff
@@ -331,9 +420,23 @@ nns_pipeline_start (nns_pipeline_h pipe)
 int
 nns_pipeline_stop (nns_pipeline_h pipe)
 {
-  /** @todo NYI */
+  nns_pipeline *p = pipe;
+  GstStateChangeReturn scret;
+  GstState state, pending;
 
-  return NNS_ERROR_NONE;
+  g_mutex_lock (&p->lock);
+  scret = gst_element_set_state (p->element, GST_STATE_PAUSED);
+  g_mutex_unlock (&p->lock);
+
+  if (scret == GST_STATE_CHANGE_SUCCESS || scret == GST_STATE_CHANGE_ASYNC)
+    return NNS_ERROR_NONE;
+
+  scret = gst_element_get_state (p->element, &state, &pending, 1000000UL);      /* 1ms */
```

Comment:
Why checking element state again after 1ms ?

Reply:
It's not checking "after 1ms". it's allowing "1ms" for state checking. If state checking is done right away at the call, it will return without delay. It only waits "1ms" only if the state is not determined during that time.

Reply:
This check is performed when scret returns with GST_STATE_CHANGE_FAILURE.
Why checking for the state change again even though `gst_element_set_state()` returned failure.